### PR TITLE
Scale Refactor

### LINKED ISSUE #2271: Scale Refactor
- [ ] QuantitativeScale._DEFAULT_NUM_TICKS seems like it is not used and should be deleted.
- [ ] Possibly remove the colorRange constructor parameter from InterpolatedColorScale.

### PR DESCRIPTION
**Pull request by aicioara**
Closes #2271

Interpolated Color scale no longer takes the colorRange as a constructor argument. Instead, it is now set via the `range()` method

**Reason:** We want InterpolatedColor to be consistent with Color, which takes as a first argument the scale type. In the same time, the argument was optional, and could have been set by the `colorRange()` method

Changelog:
- `Scales.InterpolatedColor.colorRange()` -> `Scales.InterpolatedColor.range()`
- `new Scales.InterpolatedColor(colorRange, scaleType)` -> `new Scales.InterpolatedColor(scaleType)`

**QE Notes** Regression pass on interpolated color scale. Checked the overlaying tests myself

### CODE REVIEW

**Comment by jtlan** [comment]
Should `colorRange()` just be called `range()`?

**Comment by aicioara** [comment]
I am okay with the change. This makes it more consistent with `_getRange()` which just returns `this.colorRange()`

File: src/scales/interpolatedColorScale.ts [hunk 1]
```diff
@@ -53,14 +53,11 @@ export module Scales {
...
-     * @constructor
-     * @param {string[]} [colors=InterpolatedColor.REDS] an array of strings representing color hex values
-     *   ("#FFFFFF") or keywords ("white").
      * @param {string} [scaleType="linear"] One of "linear"/"log"/"sqrt"/"pow".
      */
-    constructor(colorRange = InterpolatedColor.REDS, scaleType = "linear") {
+    constructor(scaleType = "linear") {
       super();
-      this._colorRange = colorRange;
+      this._colorRange = InterpolatedColor.REDS;
```
Comment (jtlan) [inline src/scales/interpolatedColorScale.ts]: Shouldn't we use `this.range(InterpolatedColor.REDS)`?

**Comment by jtlan** [comment]
[-1] minor.

**Pull request merged**